goal_src/jak2/engine/common_objs/projectile.gc

```diff
@@ -7,3 +7,720 @@
+           (seek! (-> self tween) 1.0 (* 3.0 (-> self clock seconds-per-frame)))
+           )
+          ((< f0-4 40960.0)
+           (seek! (-> self tween) 1.0 (-> self clock seconds-per-frame))
+           )
+          )
+        )
+      )
+    (let ((gp-0 (-> self root-override)))
+      (set! (-> self pre-move-transv quad) (-> gp-0 transv quad))
+      (let ((s5-0 (new 'stack-no-clear 'vector)))
+        (set! (-> s5-0 quad) (-> gp-0 trans quad))
+        ((-> self move) self)
+        (play-impact-sound! self)
+        (set! (-> self old-dist (-> self old-dist-count)) (* 0.0625 (vector-vector-distance s5-0 (-> gp-0 trans))))
+        )
+      )
+    (set! (-> self old-dist-count) (logand (+ (-> self old-dist-count) 1) 15))
+    (let ((f0-14 0.0))
+      (countdown (v1-38 16)
+        (+! f0-14 (-> self old-dist v1-38))
+        )
+      (if (or (and (logtest? (-> self options) (projectile-options proj-options-4))
+                   (and (logtest? (-> self root-override status) (cshape-moving-flags t-wall)) (< f0-14 2048.0))
+                   )
+              (< f0-14 (-> self stop-speed))
+              )
+          (go-sitting! self)
+          )
+      )
+    (none)
+    )
+  :code (behavior ()
+    (until #f
+      (spawn-impact-particles self)
+      (play-impact-sound self (projectile-options lose-altitude proj-options-2))
+      (suspend)
+      )
+    #f
+    (none)
+    )
+  )
+
+(defstate impact (projectile)
+  :virtual #t
+  :event projectile-event-handler
+  :enter (behavior ()
+    (spawn-shell-particles self)
+    (play-impact-sound self (projectile-options lose-altitude))
+    (none)
+    )
+  :code (behavior ()
+    (suspend)
+    (go-virtual die)
+    (none)
+    )
+  )
+
+(defstate dissipate (projectile)
+  :virtual #t
+  :event projectile-event-handler
+  :enter (behavior ()
+    (unknown-particles self)
+    (play-impact-sound self (projectile-options proj-options-2))
+    (none)
+    )
+  :code (behavior ()
+    (suspend)
+    (go-virtual die)
+    (none)
+    )
+  )
+
+(defstate die (projectile)
+  :virtual #t
+  :code (behavior ()
+    (let ((v1-0 (-> self notify-handle)))
+      (send-event (handle->process v1-0) 'notify 'die)
+      )
+    ((method-of-object self process-drawable-method-17))
+    (none)
+    )
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod init-proj-settings! projectile ((obj projectile))
+  "Init relevant settings for the [[projectile]] such as gravity, speed, timeout, etc
+       :virtual"
+  0
+  (none)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod init-collision! projectile ((obj projectile))
+  "Init the [[projectile]]'s [[collide-shape]]"
+  (let ((s5-0 (new 'process 'collide-shape-moving obj (collide-list-enum hit-by-player))))
+    (set! (-> s5-0 dynam) (copy *standard-dynamics* 'process))
+    (set! (-> s5-0 reaction)
+          (the-as (function control-info collide-query vector cshape-moving-flags) cshape-reaction-projectile)
```

Comment:
it looks like the type of `reaction` is wrong. Can you change it to `(function control-info collide-query vector vector cshape-moving-flags)`? Casting functions like this is scary because it's easy for arguments to be lost/garbage. (I made a mistake on this exact function in jak 1 and it caused the jerkiness on the lpc slide that took forever to figure out, so I'm a bit paranoid)